lib/collection/property-list.js

```diff
@@ -266,7 +272,7 @@ _.extend(PropertyList.prototype, /** @lends PropertyList.prototype */ {
      */
     map: function (iterator, context) {
         !context && (context = this);
-        return _.map(this.members, iterator, context);
+        return _.map(this.members, iterator.bind(context));
```

Comment:
@czardoz The new diff view is confusing, please check out the cumulative diff. I've made appropriate changes only where needed, excessive fallbacks to `this` via `||` are avoided.